packages/cid-verifier/src/verification.js

```diff
@@ -0,0 +1,146 @@
+/* eslint-env serviceworker, browser */
+/* global Response */
+import { getFromDenyList } from './utils/denylist'
+
+/**
+ * Get verification results from 3rd parties stored in KV.
+ *
+ * @param {string} cid
+ * @param {import('./env').Env} env
+ */
+async function getResults (cid, env) {
+  const datastore = env.CID_VERIFIER_RESULTS
+  if (!datastore) {
+    throw new Error('CID_VERIFIER_RESULTS db not ready')
+  }
+
+  return (await datastore.list({ prefix: cid }))?.keys?.reduce((acc, key) => {
+    // @ts-ignore
+    acc[key?.name] = key?.metadata?.value
+    return acc
+  }, {})
+}
+
+/**
+ * @param {Array<string>} params
+ * @param {(params: Array<string>, request: Request, env: import('./env').Env) => Promise<Response>} fn
+ * @returns {import('itty-router').RouteHandler<Request>}
+ */
+function withRequiredQueryParams (params, fn) {
+  /**
+   * @param {Request} request
+   * @param {import('./env').Env} env
+   */
+  return async function (request, env) {
+    const searchParams = (new URL(request.url)).searchParams
+
+    for (const param of params) {
+      if (!searchParams.get(param)) {
+        return new Response(`${param} is a required query param`, { status: 400 })
+      }
+    }
+
+    return await fn(params.map(param => String(searchParams.get(param))), request, env)
+  }
+}
+
+export const verificationGet = withRequiredQueryParams(['cid'],
+  /**
+   * Returns google malware result.
+   */
+  async function (params, request, env) {
+    const [cid] = params
+
+    const denyListResource = await getFromDenyList(cid, env)
+    if (denyListResource) {
+      const { status } = JSON.parse(denyListResource)
+      if (status === 451) {
+        return new Response('BLOCKED FOR LEGAL REASONS', { status: 451 })
+      } else {
+        return new Response('MALWARE DETECTED', { status: 403 })
+      }
+    }
+    return new Response('', { status: 204 })
+  }
+)
+
+/**
+ * Process CID with malware verification parties.
+ */
+export const verificationPost = withRequiredQueryParams(['cid', 'url'],
+  async function verificationPost (params, request, env) {
+    const [cid, url] = params
+    const resultKey = `${cid}/google-evaluate.result`
+    const lockKey = `${cid}/google-evaluate.lock`
+    const cidVerifyResults = await getResults(cid, env)
+    // @ts-ignore
+    const googleEvaluateResult = cidVerifyResults[resultKey]
+    // @ts-ignore
```

Comment:
it would be nice to refactor edge-gateway and cid-verifier to use `.ts` files after this.  comment annotations have limitations, like not allowing `as SomeInterface`.  i find it to be cumbersome the way it is in `.js` files...